### PR TITLE
[4.0] Improve mobile view of permissions

### PR DESCRIPTION
**Pull request by astridx**
Pull Request for Issue https://github.com/joomla/joomla-cms/pull/27615#issuecomment-578321754 .

### Summary of Changes
Show the columns in the permissions table below each other.

### Testing Instructions
1. Apply this patch and run `npm ci`.
2. Open permissions on a small display.

### My changes
Contacts  Options   test   Administration

### Actual result
Contacts  Options   test   Administration(1)

### CODE REVIEW

File: administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss [hunk 1]
```diff
@@ -246,6 +246,41 @@ joomla-tab {
...
+            display: block;
+          }
+
+          .respTable thead {
+            position: absolute;
+            top: -1111px;
+            left: -1111px;
+          }
+
+          .respTable td:before {
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss]: `::before`

File: administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss [hunk 2]
```diff
@@ -246,6 +246,41 @@ joomla-tab {
...
+            float: left;
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss]: RTL?

File: administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss [hunk 3]
```diff
@@ -246,6 +246,41 @@ joomla-tab {
...
+            padding: 0 2em 0 0;
+            content: attr(data-label) ":";
+            font-weight: bold;
+          }
+
+          .respTable td:nth-child(1) {
+            font-weight: bold;
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss]: Use variable: https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/scss/_variables.scss#L129

File: administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss [hunk 4]
```diff
@@ -246,6 +246,41 @@ joomla-tab {
...
+            position: absolute;
+            top: -1111px;
+            left: -1111px;
+          }
+
+          .respTable td:before {
+            float: left;
+            padding: 0 2em 0 0;
+            content: attr(data-label) ":";
+            font-weight: bold;
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss]: Use variable: https://github.com/joomla/joomla-cms/blob/4.0-dev/administrator/templates/atum/scss/_variables.scss#L129

File: administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss [hunk 5]
```diff
@@ -246,6 +246,41 @@ joomla-tab {
     #permissions-sliders {
       section[active] {
         padding: 0 !important;
+
+        @include media-breakpoint-down(md) {
+          .respTable {
+            text-align: right;
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss]: RTL?

File: administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss [hunk 6]
```diff
@@ -246,6 +246,41 @@ joomla-tab {
...
+          .respTable td:nth-child(1) {
+            font-weight: bold;
+          }
+
+          .respTable td:last-child {
+            border-bottom: 1em var(--atum-bg-dark-80) solid;
+          }
+
+          .oddCol {
+            background: $gray-100;
```
Comment (C-Lodder) [inline administrator/templates/atum/scss/vendor/joomla-custom-elements/joomla-tab.scss]: Use `background-color`

**Comment by astridx** [comment]
@C-Lodder Thank you.
Please see if it's OK now.

File: administrator/components/com_foos [hunk 7]
```diff
@@ -0,0 +1 @@
+/home/astrid/git/joomla-development/boilerplate/src/administrator/components/com_foos
```
Comment (alikon) [inline administrator/components/com_foos]: please don't hate me :smile:

File: components/com_foos [hunk 8]
```diff
@@ -0,0 +1 @@
+/home/astrid/git/joomla-development/boilerplate/src/components/com_foos
```
Comment (alikon) [inline components/com_foos]: again ;)

Comment (astridx) [reply components/com_foos]: I do not hate you. I hate git. Why does git always do things that I don't want? It's quite clear that I don't want to commit foo components in this PR, right?

File: administrator/components/com_foos (hunk 7 above)
Comment (astridx) [reply administrator/components/com_foos]: I do not hate you. I hate git. Why does git always do things that I don't want? It's quite clear that I don't want to commit foo components in this PR, right?

**Comment by infograf768** [comment]
This needs some more changes for RTL
Screen Shot 2020-01-30 at 12 10 46

**Comment by astridx** [comment]
Thanks to everyone who gave me tips. I have improved. This can now be tested again.
Is the` <sup>2</sup>` an error in the language file?
For me it looks like this:

تنظیمات کلی   admin   مدیریت

**Comment by infograf768** [comment]
> Is the `<sup>2</sup>` an error in the language file?

Yep, it is.

**Comment by infograf768** [comment]
I suggest some small modifications for joomla-tab.scss in order to be normalized compared to other scss.
```
    #permissions-sliders {
      section[active] {
        padding: 0 !important;

        @include media-breakpoint-down(md) {
          .respTable {
            [dir=ltr] & {
              text-align: right;
            }

            [dir=rtl] & {
              text-align: left;
            }
          }

          .respTable, .respTable thead, .respTable tbody, .respTable tr, .respTable th, .respTable td {
            display: block;
          }

          .respTable thead {
            position: absolute;
            top: -1111px;
            left: -1111px;
          }

          .respTable td::before {
            content: attr(data-label) ":";
            font-weight: $bold-weight;

            [dir=ltr] & {
              float: left;
              padding: 0 2em 0 0;
            }

            [dir=rtl] & {
              float: right;
              padding: 0 0 0 2em;
            }
          }

          .respTable td:nth-child(1) {
            font-weight: $bold-weight;
          }

          .respTable td:last-child {
            border-bottom: 1em var(--atum-bg-dark-80) solid;
          }

          .oddCol {
            background: var(--atum-bg-light);
          }
        }
      }
    }
  }
}
```

**Comment by astridx** [comment]
@infograf768 I have changed it. I hope I've seen everything.

**Comment by infograf768** [comment]
looks ok, will test tomorrow. :)

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on f41a4a06bac68e3b1ac8e527b51ffd2f8f0f7f0c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27670">issues.joomla.org/tracker/joomla-cms/27670</a>.</sub>

**Comment by N6REJ** [comment]
I have tested this item :white_check_mark: successfully on f41a4a06bac68e3b1ac8e527b51ffd2f8f0f7f0c<br /><br />works via pt 4 ( ci on )<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27670">issues.joomla.org/tracker/joomla-cms/27670</a>.</sub>
pt4

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27670">issues.joomla.org/tracker/joomla-cms/27670</a>.</sub>

**Comment by infograf768** [comment]
Thanks.

**Pull request merged**